tests/ui/traits/non_lifetime_binders/on-rpit.rs

```diff
@@ -1,11 +1,14 @@
 //@ check-pass
 
+#![feature(sized_hierarchy)]
 #![feature(non_lifetime_binders)]
 //~^ WARN the feature `non_lifetime_binders` is incomplete
 
-trait Trait<T: ?Sized> {}
+use std::marker::PointeeSized;
 
-impl<T: ?Sized> Trait<T> for i32 {}
+trait Trait<T: PointeeSized> {}
+
+impl<T: PointeeSized> Trait<T> for i32 {}
 
 fn produce() -> impl for<T> Trait<T> {
```

Comment:
After the sized hierarchy is introduced, does this _need_ `sized_hierarchy` enabled to work? If yes, that is a potential concern. If not, these tests should remain without using the feature, while you can add the tests to under the experimental feature.

Reply:
It is necessary for the `non_lifetime_binders` tests - these tests only work when `for<T>` is unconstrained and with that feature being incomplete, it isn't possible to  add the bounds necessary to keep this test passing w/out the parameters being unconstrained. The only way to make a parameter unconstrained now is with `PointeeSized`, so the feature needs to be enabled.

Reply:
Understood, thanks.